src/com/comeet/data/DataRepository.java

```diff
@@ -97,10 +103,65 @@ public Room retrieveRoomMetadata(String email) {
 
         } catch (Exception e) {
             // e.printStackTrace();
-            ApiLogger.logger.log(Level.SEVERE, "Error getting rooms from database database", e);
+            //ApiLogger.logger.log(Level.SEVERE, "Error getting rooms from database database", e);
+            throw e;
         }
 
         return roomMetadata;
     }
+    
+    /**
+     * Retrieves the search criteria from the database based on the domain
```

Comment:
This is beyond the scope of this change I think.  We should talk about and document our api structure at our next meeting.